modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_test_case.ftl

```diff
@@ -1272,8 +1272,58 @@ public abstract class Base${schemaName}ResourceTestCase {
 						assertValid(get${schemaName}, multipartFiles);
 					</#if>
 				</#if>
+				<#if javaMethodSignature.methodName?cap_first?ends_with("ByExternalReferenceCode") >
+					${schemaName} new${schemaName} = test${javaMethodSignature.methodName?cap_first}_create${schemaName}();
+
+                    put${schemaName} = ${schemaVarName}Resource.${javaMethodSignature.methodName}(
+
+                    <#list javaMethodSignature.javaMethodParameters as javaMethodParameter>
+                        <#if freeMarkerTool.isPathParameter(javaMethodParameter, javaMethodSignature.operation) && stringUtil.equals(javaMethodParameter.parameterName, schemaVarName + "Id")>
+							new${schemaName}.getId()
+                        <#elseif freeMarkerTool.isPathParameter(javaMethodParameter, javaMethodSignature.operation) && properties?keys?seq_contains(javaMethodParameter.parameterName)>
+							new${schemaName}.get${javaMethodParameter.parameterName?cap_first}()
+                        <#elseif stringUtil.equals(javaMethodParameter.parameterName, "multipartBody") || stringUtil.equals(javaMethodParameter.parameterName, schemaVarName)>
+							new${schemaName}
+                        <#else>
+							null
+                        </#if>
+                        <#sep>, </#sep>
+                    </#list>
+
+                    <#if freeMarkerTool.hasRequestBodyMediaType(javaMethodSignature, "multipart/form-data")>
+						, getMultipartFiles()
+                    </#if>
+
+					);
+
+					assertEquals(new${schemaName}, put${schemaName});
+					assertValid(put${schemaName});
+
+                    get${schemaName} = ${schemaVarName}Resource.${javaMethodSignature.methodName?replace("put", "get")}(
+                    <#list javaMethodSignature.pathJavaMethodParameters as javaMethodParameter>
```

Comment:
All this indentation looks fishy in github but it's Ok, I don't know how to indent freemarker files either.